pkg/detector/ospkg/oracle/oracle.go

```diff
@@ -65,9 +77,17 @@ func (s *Scanner) Detect(osVer string, pkgs []ftypes.Package) ([]types.DetectedV
 		installed := utils.FormatVersion(pkg)
 		installedVersion := version.NewVersion(installed)
 		for _, adv := range advisories {
-			// Skip if only one of them contains .ksplice1.
-			if strings.Contains(adv.FixedVersion, ".ksplice1.") != strings.Contains(pkg.Release, ".ksplice1.") {
+			// when one of them doesn't have ksplice, we'll also skip it
+			// extract kspliceX and compare it with kspliceY in advisories
+			// if kspliceX and kspliceY are different, we will skip the advisory
+			if isKSlice := strings.Contains(adv.FixedVersion, ".ksplice"); isKSlice != strings.Contains(pkg.Release, ".ksplice") {
 				continue
+			} else if isKSlice {
+				releaseV := extractKspliceNumber(adv.FixedVersion)
+				fixedV := extractKspliceNumber(pkg.Release)
+				if releaseV != fixedV {
+					continue
+				}
```

Comment:
I think we can simplify it.

```suggestion
			if extractKspliceNumber(adv.FixedVersion) != extractKspliceNumber(pkg.Release) {
				continue
			}
```

Reply:
sure, your was is better!